metrics/influxdb.py

```diff
@@ -0,0 +1,139 @@
+        return dct
+
+
+    def serialize(self):
+        """Encodes the Point in InfluxDB Line Protocol syntax."""
+        if not self.fields:
+            raise ValueError('Point must have at least one field.\n')
+
+        def value_repr(val):
+            """Encodes a field value."""
+            if isinstance(val, bool):
+                return str(val)
+            if isinstance(val, basestring):
+                return '\"%s\"' % val.replace('"', '\"')
+            return repr(val)
+
+        def measure_repr(measurement):
+            """Encodes a measurement name."""
+            return measurement.replace(',', r'\,').replace(' ', r'\ ')
+
+        def label_repr(label):
+            """Encodes a tag name, tag value, or field name."""
+            return label.replace(',', r'\,').replace(' ', r'\ ').replace('=', r'\=')
+
+        tags = ''.join([
+            ',%s=%s' % (label_repr(key), label_repr(val))
+            for key, val in self.tags.iteritems()
+        ])
+        fields = ','.join([
+            '%s=%s' % (label_repr(key), value_repr(val))
+            for key, val in self.fields.iteritems()
+        ])
+        return '%s%s %s%s' % (
+            measure_repr(self.measurement),
+            tags,
+            fields,
+            ' %d' % self.time if self.time else ''
+        )
+
+
+class Pusher(object):
+    """Client that pushes Point objects to an InfluxDB."""
+    def __init__(self, host_port, user, password):
+        self.host_port = host_port
+        self.user = user
+        self.password = password
+
+    @staticmethod
+    def from_config(config_path):
+        """Creates an Pusher for the InfluxDB described by a json config."""
+        with open(config_path) as config_file:
+            config = json.load(config_file)
+
+        def check_config(field):
+            if not field in config:
+                raise ValueError('Pusher config requires field \'%s\'' % field)
+        check_config('hostport')
+        check_config('user')
+        check_config('password')
+        return Pusher(config['hostport'], config['user'], config['password'])
+
+    def push(self, points, database):
+        """Pushes time series data points to an InfluxDB in batches."""
+        params = urllib.urlencode(
+            {'db': database, 'u': self.user, 'p': self.password, 'precision': 's'}
+        )
+
+        stamp = int(time.time())
+        for point in points:
+            if not point.time:
+                point.time = stamp
+
+        while points:
+            body = '\n'.join(p.serialize() for p in points[:100])
+            points = points[100:]
+            for attempt in range(5):
+                if attempt:
+                    time.sleep(2 ** (attempt - 1))
+
+                try:
+                    conn = httplib.HTTPConnection(self.host_port)
+                    conn.request('POST', '/write?%s' % params, body)
+                    resp = conn.getresponse()
+                except httplib.HTTPException:
```

Comment:
Consider logging something in these ephemeral error scenarios `print >>sys.stderr, traceback.format_exc()  # or traceback.print_exc()`